packages/cloudflare/src/cli/templates/images.ts

```diff
@@ -22,16 +28,28 @@ export function fetchImage(fetcher: Fetcher | undefined, imageUrl: string) {
   // Local
   if (imageUrl.startsWith("/")) {
     let pathname: string;
+    let url: URL;
     try {
-      const url = new URL(imageUrl, "http://n");
+      // We only need pathname and search
+      url = new URL(imageUrl, "http://n");
       pathname = decodeURIComponent(url.pathname);
     } catch {
       return getUrlErrorResponse();
     }
+
     if (/\/_next\/image($|\/)/.test(pathname)) {
       return getUrlErrorResponse();
     }
 
+    // If localPatterns are not defined all local images are allowed.
+    if (__IMAGES_LOCAL_PATTERNS__.length === 0) {
+      return fetcher?.fetch(`http://assets.local${imageUrl}`);
+    }
+
+    if (!__IMAGES_LOCAL_PATTERNS__.some((p: LocalPattern) => matchLocalPattern(p, url))) {
+      return getUrlErrorResponse();
+    }
```

Comment:
```suggestion
    if (__IMAGES_LOCAL_PATTERNS__.length > 0 && !__IMAGES_LOCAL_PATTERNS__.some((p: LocalPattern) => matchLocalPattern(p, url))) {
      return getUrlErrorResponse();
    }
```

to avoid repeating `return fetcher?.fetch(`http://assets.local${imageUrl}`);` - if that piece need to be updated at some point, we will probably miss one of the 2